collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -1255,7 +1055,234 @@ public void onClick(View v) {
+				Log.e(t, e.getMessage(), e);
+				createErrorDialog(e.getMessage() + "\n\n" + e.getCause().getMessage(), DO_NOT_EXIT);
+			}
+		}
+
+        // change start screen based on navigation prefs
+        String navigationChoice = getPreferences().getString(
+									PreferencesActivity.KEY_NAVIGATION,
+									PreferencesActivity.KEY_NAVIGATION);
+
+        ImageView imageAdvance = ((ImageView) startView.findViewById(R.id.image_advance));
+        ImageView imageBackup = ((ImageView) startView.findViewById(R.id.image_backup));
+        TextView textAdvance = ((TextView) startView.findViewById(R.id.text_advance));
+        TextView textBackup = ((TextView) startView.findViewById(R.id.text_backup));
+        TextView description = ((TextView) startView.findViewById(R.id.description));
+
+		Boolean useSwipe = false;
+		Boolean useButtons = false;
+
+		if (navigationChoice != null) {
+            if (navigationChoice.contains(PreferencesActivity.NAVIGATION_SWIPE)) {
+                useSwipe = true;
+            }
+
+            if (navigationChoice.contains(PreferencesActivity.NAVIGATION_BUTTONS)) {
+                useButtons = true;
+            }
+        }
+
+        if (useSwipe && !useButtons) {
+            description.setText(getString(R.string.swipe_instructions, formController.getFormTitle()));
+        } else if (useButtons && !useSwipe) {
+            imageAdvance.setVisibility(View.GONE);
+            imageBackup.setVisibility(View.GONE);
+            textAdvance.setVisibility(View.GONE);
+            textBackup.setVisibility(View.GONE);
+            description.setText(getString(R.string.buttons_instructions, formController.getFormTitle()));
+        } else {
+            description.setText(getString(R.string.swipe_buttons_instructions, formController.getFormTitle()));
+        }
+
+        if (mBackButton.isShown()) {
+            mBackButton.setEnabled(false);
+        }
+        if (mNextButton.isShown()) {
+            mNextButton.setEnabled(true);
+        }
+
+		// Only show intro screen once, flip boolean flag so it wont be shown again
+		getPreferences().edit().putBoolean(PreferencesActivity.KEY_SHOW_INTRO, false).apply();
+
+        return startView;
+    }
+
+	private SharedPreferences getPreferences() {
+		return PreferenceManager.getDefaultSharedPreferences(this);
+	}
+
+	private boolean shouldShowIntroScreenBecauseOfSettings() {
+		return getPreferences().getBoolean(PreferencesActivity.KEY_SHOW_INTRO, true);
+	}
+
+	private boolean shouldShowIntroScreenBecauseOfLogo(FormController formController){
+		return getFormLogo(formController) != null;
+	}
+
+	private Drawable getFormLogo(FormController formController) {
+        Drawable image = null;
+        File mediaFolder = formController.getMediaFolder();
+        String mediaDir = mediaFolder.getAbsolutePath();
+        BitmapDrawable bitImage = null;
+        // attempt to load the form-specific logo...
+        // this is arbitrarily silly
```

Comment:
This kind of comment would be good to address and/or remove (the "arbitrarily silly" part).